src/broadcast/scrobblingmanager.cpp

```diff
@@ -0,0 +1,210 @@
+                    qDebug() << "Track doesn't contain player"
+                                "yet is requested for deletion.";
+                    break;
+                }
+                //Load error, stray from engine buffer.
+                BaseTrackPlayer *player = 
+                    m_pManager->getPlayer(candidateTrack.m_playerGroup);
+                if (player->getLoadedTrack() ==
+                    candidateTrack.m_pTrack) 
+                    break;                    
+                QLinkedList<QString>::iterator it = 
+                    trackInfo->m_players.begin();
+                while (it != trackInfo->m_players.end() &&
+                       *it != candidateTrack.m_playerGroup) {                    
+                    ++it;
+                }
+                if (*it == candidateTrack.m_playerGroup) {
+                    trackInfo->m_players.erase(it);
+                }
+                if (trackInfo->m_players.empty()) {
+                    trackInfo->m_trackInfo.pausePlayedTime();
+                    trackInfo->m_trackInfo.resetPlayedTime();
+                    delete trackInfo;
+                    m_trackList.erase(trackListIterator);
+                }
+                break;  
+            }
+        }
+    }
+}
+
+bool ScrobblingManager::isTrackAudible(TrackPointer pTrack, BaseTrackPlayer * pPlayer) {
+    if (pPlayer->getLoadedTrack() != pTrack) {
+        qDebug() << "Track can't be audible because is not in player";
+        return false;
+    }   
+    return getPlayerVolume(pPlayer) >= 0.20;
+}
+
+double ScrobblingManager::getPlayerVolume(BaseTrackPlayer *pPlayer) {
+    double finalVolume;
+    ControlProxy trackPreGain(pPlayer->getGroup(),"pregain",this);
+    double preGain = trackPreGain.get();
+    ControlProxy trackVolume(pPlayer->getGroup(),"volume",this);
+    double volume = trackVolume.get();
+    ControlProxy deckOrientation(pPlayer->getGroup(),"orientation",this);
+    int orientation = deckOrientation.get();    
+
+    double xFaderLeft,xFaderRight;
+
+    EngineXfader::getXfadeGains(m_CPCrossfader.get(),
+                                m_CPXFaderCurve.get(),
+                                m_CPXFaderCalibration.get(),
+                                m_CPXFaderMode.get(),
+                                m_CPXFaderReverse.toBool(),
+                                &xFaderLeft,&xFaderRight);
+    finalVolume = preGain * volume;
+    if (orientation == EngineChannel::LEFT)
+        finalVolume *= xFaderLeft;
+    else if (orientation == EngineChannel::RIGHT) 
+        finalVolume *= xFaderRight;
+    return finalVolume;
+}
+
+void ScrobblingManager::slotGuiTick(double timeSinceLastTick) {
+    for (TrackInfo *trackInfo : m_trackList) {
+        trackInfo->m_trackInfo.slotGuiTick(timeSinceLastTick);
+    }
+}
+
+void ScrobblingManager::timerEvent(QTimerEvent *timerEvent) {
+    for (TrackInfo *trackInfo : m_trackList) {
+        bool inaudible = true;
```

Comment:
Please try to avoid naming a boolean variables with a negation names. The name "audible" would work much better here. But the local variable is not needed anyway, the code can be simplified.

Reply:
I'll change the name of the variable.

Reply:
You don't need the variable.

Reply:
Can you elaborate?